jujugui/static/gui/src/app/app.js

```diff
@@ -1011,11 +1011,82 @@ YUI.add('juju-gui', function(Y) {
     },
 
     /**
+      Get user name from cookie.
+
+      @method _getUsernameFromCookie
+    */
+    _getUsernameFromCookie: function() {
+      // XXX jcsackett 2016-01-08: We should be using the whoami endpoint on
+      // charmstore to get the username of a logged in jem user rather than
+      // parsing the cookie. Consider this all a bad hack for demo and we'll do
+      // something better in the very near future.
+
+      // Don't look for cookies if in sandbox.
+      if (this.get('sandbox')) {
+        return null;
+      }
+      var cookie = document.cookie;
+      var data = cookie.split(';');
+      var macaroon = null;
+      var prefix = 'Macaroons-jem=';
+      var found = data.some(function(item) {
+        macaroon = item;
+        return (item.indexOf(prefix) !== -1);
```

Comment:
-1 indicates nothing was found, so this says the prefix was found. We can't just look for an instance of prefix being found at 0, since there can be an unknown amount of leading space.